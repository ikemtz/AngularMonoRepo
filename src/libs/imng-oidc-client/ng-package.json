{
  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../dist/libs/imng-oidc-client",
  "lib": {
    "entryFile": "src/index.ts"
  },
  "assets": ["./src/assets/*"]
}
