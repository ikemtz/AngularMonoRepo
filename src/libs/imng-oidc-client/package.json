{
  "name": "imng-oidc-client",
  "version": "0.0.1",
  "description": "Small library to facilitate the integration of the ng-oidc-client library with the Auth0 identity provider.",
  "peerDependencies": {
    "@angular/common": ">=13.x",
    "@angular/core": ">=13.x",
    "@angular/router": ">=13.x",
    "@ngrx/store": ">=13.x",
    "@ngrx/effects": ">=13.x",
    "imng-ngrx-utils": "*",
    "imng-nrsrx-client-utils": "*",
    "oidc-client": ">=1.x",
    "rxjs": "*"
  },
  "dependencies": {
    "tslib": "^2.3.0"
  },
  "keywords": [
    "Angular",
    "oidc-client",
    "ngrx"
  ],
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "https://github.com/ikemtz/AngularMonoRepo.git"
  },
  "icon": "https://avatars.githubusercontent.com/u/6444182?v=4",
  "private": false,
  "author": {
    "name": "Isaac Martinez <@ikemtz>",
    "url": "https://github.com/ikemtz"
  }
}
