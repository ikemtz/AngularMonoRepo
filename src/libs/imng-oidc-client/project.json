{
  "name": "imng-oidc-client",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "projectType": "library",
  "sourceRoot": "libs/imng-oidc-client/src",
  "prefix": "imng",
  "tags": [],
  "targets": {
    "build": {
      "executor": "@nx/angular:package",
      "outputs": ["{workspaceRoot}/dist/libs/imng-oidc-client"],
      "dependsOn": [
        {
          "target": "build",
          "dependencies": true
        }
      ],
      "options": {
        "project": "libs/imng-oidc-client/ng-package.json"
      },
      "configurations": {
        "production": {
          "tsConfig": "libs/imng-oidc-client/tsconfig.lib.prod.json"
        },
        "development": {
          "tsConfig": "libs/imng-oidc-client/tsconfig.lib.json"
        }
      },
      "defaultConfiguration": "production"
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/imng-oidc-client"],
      "options": {
        "jestConfig": "libs/imng-oidc-client/jest.config.ts"
      }
    },
    "lint": {
      "executor": "@nx/eslint:lint"
    }
  }
}
