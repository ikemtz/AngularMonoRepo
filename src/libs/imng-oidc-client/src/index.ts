export * from './lib/imng-oidc-client.module';
export * from './lib/+state/oidc-user.facade';
export { OidcUserSelectors } from './lib/+state/oidc-user.selectors';
export * from './lib/+state/oidc.actions';
export * from './lib/+state/oidc.effects';
export * from './lib/+state/oidc.facade';
export { OidcState, ErrorState, oidcReducer } from './lib/+state/oidc.reducer';
export { oidcQuery } from './lib/+state/oidc.selectors';
export * from './lib/models/arguments.model';
export {
  OidcClientConfig,
  OIDC_CLIENT_CONFIG,
} from './lib/models/oidc-client-config';
export * from './lib/models/oidc-user-profile';
export * from './lib/models/constants';
export * from './lib/util/oidc-client-configurator';
export * from './lib/services/auth-guard';
export * from './lib/services/oidc.service';
export * from './lib/services/token-interceptor.service';
export * from './lib/util/jwt-decoder';
export * from './lib/services/permissions.guard';
export * from './lib/components/access-denied.component';
export * from './lib/support/support.component';
export * from './lib/img-oidc-client-routing.module';
export * from './lib/models/oidc-user';
export * from './lib/components/logout-success.component';
