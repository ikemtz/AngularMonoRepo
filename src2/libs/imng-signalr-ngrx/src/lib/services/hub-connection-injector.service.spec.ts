import { TestBed } from '@angular/core/testing';

import { HubConnectionInjectorService } from './hub-connection-injector.service';
import { SIGNALR_CONFIG } from '../models/signalr.configuration';
import { OidcFacade } from 'imng-oidc-client';
import { of } from 'rxjs';
import { Store } from '@ngrx/store';

describe('HubConnectionInjectorService', () => {
  let service: HubConnectionInjectorService;

  beforeEach(() => {
    TestBed.configureTestingModule({
      providers: [
        {
          provide: SIGNALR_CONFIG,
          multi: false,
          useValue: { hostUrl: 'http://xyz/notificationHub', logLevel: 1 },
        },
        { provide: OidcFacade, useValue: { accessToken$: of('xyz') } },
        { provide: Store, useValue: { dispatch: jest.fn() } },
      ],
    });
    service = TestBed.inject(HubConnectionInjectorService);
  });

  it('should be created', () => {
    expect(service).toBeTruthy();
    expect(
      (service as unknown as { allSubscriptions: [] }).allSubscriptions.length
    ).toBe(1);
  });

  it('should be ngOnDestroy', () => {
    service.ngOnDestroy();
  });
});
