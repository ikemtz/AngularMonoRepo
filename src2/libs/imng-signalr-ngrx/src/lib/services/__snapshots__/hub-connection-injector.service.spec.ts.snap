// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HubConnectionInjectorService should throw an error on improper host 1`] = `[Error: WebSocket failed to connect. The connection could not be found on the server, either the endpoint may not be a SignalR endpoint, the connection ID is not present on the server, or there is a proxy blocking WebSockets. If you have multiple servers check that sticky sessions are enabled.]`;
